app/models/publisher.rb

```diff
@@ -19,6 +19,27 @@ def publish
     end
   end
 
+  def discard_draft
+    begin
+      ActiveRecord::Base.transaction do
+        publishing_api.discard_draft(content_model.content_id)
+
+        latest_published_edition = content_model.editions.published.last
+        if latest_published_edition.present?
+          content_model
+            .editions
+            .where("created_at > ?", latest_published_edition.created_at)
+            .destroy_all
+        else
+          content_model.destroy!
+        end
+      end
+      DiscardDraftResponse.new(success: true)
+    rescue GdsApi::HTTPErrorResponse => e
+      DiscardDraftResponse.new(success: false, errors: e.error_details['error']['message'])
+    end
```

Comment:
If we rename `PublicationResponse` with a name you'd feel comfortable using here then this exception rescuing code is shared and could be reused.